app/models/manageiq/providers/redhat/infra_manager/api_integration.rb

```diff
@@ -233,7 +233,7 @@ def verify_credentials(args)
         :password         => ManageIQ::Password.try_decrypt(password),
         :server           => server,
         :port             => port,
-        :verify_ssl       => verify_ssl ? OpenSSL::SSL::VERIFY_PEER : OpenSSL::SSL::VERIFY_NONE,
+        :verify_ssl       => verify_ssl,
```

Comment:
I think this interface used to take either an integer (that is an OpenSSL::VERIFY* constant) or a boolean.  Do we still want the backend verify method to support both or do we only want to support integers?

Reply:
also, the ruby internal code handles nil in a strange way in terms of verifying the peer.

Reply:
Maybe?  Do you mean this code in particular or `self.verify_credentials` in general

This code looks like if verify_ssl is an int it will always `VERIFY_PEER`

Reply:
> also, the ruby internal code handles nil in a strange way in terms of verifying the peer.

I can `verify_ssl ||=` to ensure this is never nil if that'd help, these args come from the DDF params_for_create which has a default value and so this shouldn't ever be nil (famous last words of course)

Reply:
I meant verify_credentials in general.  I thought we had sort of casually of agreed on supporting both a boolean (for convenience, e.g. in rails console) and a integer (for accuracy, e.g. via DDF).  I'm fine with any, just thought it made more sense to decide something now and then ultimately ensure the interface is consistent.

Reply:
> This code looks like if verify_ssl is an int it will always VERIFY_PEER

Yeah the original code is written as the boolean case, and apparently didn't ever support integers so it is wrong regardless.  I'm more trying to understand what is interface we _want_ to support.  I had thought we wanted both boolean and integer, which is why this changed jumped out at me, cause I was expecting something roughly more like:

```ruby
        verify_ssl = verify_ssl ? OpenSSL::SSL::VERIFY_PEER : OpenSSL::SSL::VERIFY_NONE unless verify_ssl.is_a?(Integer)
        :verify_ssl  => verify_ssl
```

Reply:
Actually, looking higher up the method, it seems this has always come out of the default_endpoint, and will always be an integer.  I didn't realize this particular method `verify_credentials` was reworked to be _only_ DDF based (was surprised to see the args don't allow for overwriting the value, but the args has completely changed). Kind of wish this method was named verify_credentials_ddf :(

Sorry for the tangent, but this is fine.